modules/reactor/src/meshgenerators/CoarseMeshExtraElementIDGenerator.C

```diff
@@ -53,11 +57,32 @@ CoarseMeshExtraElementIDGenerator::generate()
   std::unique_ptr<MeshBase> mesh = std::move(_input);
 
   unsigned int coarse_id;
-  if (!mesh->has_elem_integer(_coarse_id_name))
+  const bool already_has_id = mesh->has_elem_integer(_coarse_id_name);
+  if (!already_has_id)
     coarse_id = mesh->add_elem_integer(_coarse_id_name);
   else
     coarse_id = mesh->get_elem_integer_index(_coarse_id_name);
 
+  // If we are not going through the entire mesh and the ID already exists, we need to offset the
+  // assigned ID so that we don't copy ones that are already there
+  dof_id_type id_offset = 0;
+  if (already_has_id && isParamValid("subdomains"))
+  {
+    for (auto & elem : mesh->active_element_ptr_range())
+    {
+      dof_id_type elem_id = elem->get_extra_integer(coarse_id);
+      if (elem_id != DofObject::invalid_id && elem_id > id_offset)
+        id_offset = elem_id;
+    }
+    id_offset++;
+  }
+
+  // Get the requested subdomain IDs
+  std::set<SubdomainID> include_subdomains;
+  for (const auto & sid :
+       MooseMeshUtils::getSubdomainIDs(*mesh, getParam<std::vector<SubdomainName>>("subdomains")))
```

Comment:
does this fail when the subdomain is not in the mesh ? (due to a typo for example?)

Reply:
ok it doesnt it just gives invalid_subdomain id

we need a check that are subdomain names passed are in the mesh